openmls/src/treesync/tests_and_kats/tests/test_diff.rs

```diff
@@ -38,15 +41,15 @@ fn test_free_leaf_computation(ciphersuite: Ciphersuite, backend: &impl OpenMlsCr
     // Build a rudimentary tree with two populated and two empty leaf nodes.
     let nodes: Vec<Option<Node>> = vec![
         Some(Node::LeafNode(
-            kpb_0.key_package().leaf_node().clone().into(),
+            LeafNode::<StateValidated>::from(kpb_0.key_package().leaf_node().clone()).into(),
```

Comment:
This can be simplified. The `StateValidated` serves as a umbrella type currently that can be used after any of `StateKeyPackage, StateAddUpdateCommit, StateRatchet`. I'm not happy with it and would like to improve this.